lib/Sabberworm/CSS/CSSList/CSSList.php

```diff
@@ -142,6 +142,13 @@ private static function parseAtRule(ParserState $oParserState) {
 		} else {
 			//Unknown other at rule (font-face or such)
 			$sArgs = trim($oParserState->consumeUntil('{', false, true));
+			if (substr_count($sArgs, "(") != substr_count($sArgs, ")")) {
+				if($oParserState->getSettings()->bLenientParsing) {
+					return NULL;
```

Comment:
Will this mean there’s a `null` item in the resulting document’s contents? Or does the calling code ignore returned `null`s?

Reply:
There are no `null` elements in the resulting document. The calling code looks like this:
```PHP
			$oListItem = null;
			if($bLenientParsing) {
				try {
					$oListItem = self::parseListItem($oParserState, $oList);
				} catch (UnexpectedTokenException $e) {
					$oListItem = false;
				}
			} else {
				$oListItem = self::parseListItem($oParserState, $oList);
			}
			if($oListItem === null) {
				// List parsing finished
				return;
			}
```
Returning `null` aborts further parsing, which is the desired behavior. Otherwise the resulting CSS will (most likely) be valid which will change the final rendering of pages with this issue.

Reply:
Ok, I get it now, thanks…

Reply:
I presume you meant “resulting CSS will (most likely) be _in_valid”

Reply:
I meant "valid", in the sense that the broken rules will be discarded, but the rest of the rules, which I presume are valid, will be included in the file.

However, this will be an issue, since those rules have been previously ignored by the browsers, so including them will most likely alter the final rendering of the page.